module/PowerShellEditorServices/Commands/Public/Import-EditorCommand.ps1

```diff
@@ -59,7 +59,7 @@ function Import-EditorCommand {
         }
         $flags = [Reflection.BindingFlags]'Instance, NonPublic'
         $extensionService = $psEditor.GetType().
-                                      GetField('extensionService', $flags).
+                                      GetField('_extensionService', $flags).
                                       GetValue($psEditor)
 
         $editorCommands = $extensionService.GetType().
```

Comment:
Actually you can replace all of these with `$editorCommands = $psEditor.GetCommands()`.  That API didn't exist at the time, but now that it does that should definitely be used instead.

Reply:
Ah yeah that looks a lot nicer. I can change it to that but I'm not sure about the correct procedure in terms of not cluttering up the commit history - should I do an interactive rebase to clean up or leave it up to the maintainer to squash & merge?

Reply:
> leave it up to the maintainer to squash & merge?

That one :)

Also taking a closer look you'll probably want to do something like:

```powershell
$editorCommands = [System.Collections.Generic.Dictionary[string, Microsoft.PowerShell.EditorServices.Services.PowerShellContext.EditorCommand]]::new(
    [System.StringComparer]::OrdinalIgnoreCase)

foreach ($existingCommand in $psEditor.GetCommands()) {
    $editorCommands.Add($existingCommand.Name, $existingCommand)
}
```

Reply:
Thanks for the pointers - will get that implemented and confirm it all still works

Out of interest why do you suggest a strongly typed dictionary over a regular hashtable? Performance or just clearer intent/more control or something?

Reply:
Actually I was just matching the type of the field, `hashtable` will probably work just fine and be a bit more clean.  Good call :)

Reply:
Awesome. Just pushed new version 🙂